src/coreclr/jit/inlinepolicy.cpp

```diff
@@ -1058,41 +1163,95 @@ double DefaultPolicy::DetermineMultiplier()
         //   typeof(T1) == typeof(T2)
         //   Math.Abs(constArg)
         //   BitOperation.PopCount(10)
-        JITDUMP("\nInline has %d foldable intrinsics.", m_FoldableIntrinsic);
+        multiplier += 3.0 + m_FoldableIntrinsic;
+        JITDUMP("\nInline has %d foldable intrinsics.  Multiplier increased to %g.", m_FoldableIntrinsic, multiplier);
     }
 
     if (m_FoldableExpr > 0)
     {
         // E.g. add/mul/ceq, etc. over constant/constant arguments
-        JITDUMP("\nInline has %d foldable binary expressions.", m_FoldableExpr);
+        multiplier += 2.0 + m_FoldableExpr;
+        JITDUMP("\nInline has %d foldable binary expressions.  Multiplier increased to %g.", m_FoldableExpr,
+                multiplier);
     }
 
     if (m_FoldableExprUn > 0)
     {
         // E.g. casts, negations, etc. over constants/constant arguments
-        JITDUMP("\nInline has %d foldable unary expressions.", m_FoldableExprUn);
+        multiplier += m_FoldableExpr;
+        JITDUMP("\nInline has %d foldable unary expressions.  Multiplier increased to %g.", m_FoldableExprUn,
+                multiplier);
     }
 
     if (m_DivByCns > 0)
     {
         // E.g. callee has "x / arg0" where arg0 is a const at the call site -
         // we'll avoid a very expensive DIV instruction after inlining.
-        JITDUMP("\nInline has %d Div-by-constArg expressions.", m_DivByCns);
+        // multiplier += 3.0 + m_DivByCns;
+        JITDUMP("\nInline has %d Div-by-constArg expressions.  Multiplier increased to %g.", m_DivByCns, multiplier);
     }
 
-    if (m_BackwardJump)
+    if (m_HasProfile && (m_CallsiteFrequency != InlineCallsiteFrequency::RARE))
```

Comment:
Does this rare exclusion still make sense? Seems like if the call site frequency is low enough it could naturally suppress inlining, if the multiplier value was < 1.0?